lib/src/model/nutriments.dart

```diff
@@ -122,3 +194,71 @@ class Nutriments extends JsonObject {
 
   static Map<String, dynamic> toJsonHelper(Nutriments? n) => n?.toJson() ?? {};
 }
+
+class NutrimentValue {
+  final double? value;
+  final NutrimentModifier? modifier;
+
+  NutrimentValue({this.value, this.modifier})
+      : assert(
+          modifier == NutrimentModifier.noValue ? value == null : true,
+          'Value should be null when modifier is noValue',
+        ),
+        assert(
+          [
+            NutrimentModifier.moreThan,
+            NutrimentModifier.lessThan,
+            NutrimentModifier.approximately
+          ].contains(modifier)
+              ? value != null
+              : true,
+          'Value should not be null when modifier is moreThan, lessThan or approximately',
+        );
+
+  /// No value provided means no value on the packaging.
+  /// And it's not the same as the user doesn't know the value.
+  NutrimentValue.noValueProvided()
+      : value = null,
+        modifier = NutrimentModifier.noValue;
+
+  NutrimentValue._fromMap(Map<String, dynamic> map, String tag)
+      : this(
+          value: map[tag],
+          modifier: NutrimentModifier.fromValue(map['${tag}_modifier']),
+        );
+
+  /// You can format the data yourself, or use this method
+  String? get formattedValue {
+    if (modifier == NutrimentModifier.noValue) {
+      return modifier!.value;
+    } else if (modifier != null) {
+      return '${modifier!.value} $value';
+    } else {
+      return value?.toString();
+    }
+  }
+}
+
+enum NutrimentModifier {
```

Comment:
* Should be spelled `NutrientModifier`
* Could implement `OffTagged`
* Not sure if we need `@JsonValue` here, which is redundant with the `value` field
* Would rename field `value` as `modifier`